service/src/main/java/tds/session/repositories/impl/SessionRepositoryImpl.java

```diff
@@ -40,7 +43,22 @@ public SessionRepositoryImpl(final NamedParameterJdbcTemplate jdbcTemplate) {
 
     @Override
     public Optional<Session> findSessionById(final UUID id) {
-        final SqlParameterSource parameters = new MapSqlParameterSource("id", UuidAdapter.getBytesFromUUID(id));
+        List<Session> sessions = findSessionsByIds(id);
+
+        if (sessions.size() > 0) {
```

Comment:
And can we move this into the service layer and remove this method from the repository?